### PR TITLE
[Resource][Grid] Add possibility to configure parameters for action

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | yes |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/Grid/Parser/OptionsParser.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+     */
+    private function parseOptionResourceField($value, Request $request, $data)
+    {
+        $value = preg_replace_callback('/(\$\w+)/', function ($matches) use ($request) {
+            $variable = $request->get(substr($matches[1], 1));
+
+            return is_string($variable) ? sprintf('"%s"', $variable) : $variable;
+        }, $value);
+
+        $value = 'get'.ucfirst($value);
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/Grid/Parser/OptionsParser.php]: We should not do that, value will be enough. :)

File: src/Sylius/Bundle/ResourceBundle/Grid/Parser/OptionsParser.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+     * @param mixed $data
+     *
+     * @return string
+     */
+    private function parseOptionResourceField($value, Request $request, $data)
+    {
+        $value = preg_replace_callback('/(\$\w+)/', function ($matches) use ($request) {
+            $variable = $request->get(substr($matches[1], 1));
+
+            return is_string($variable) ? sprintf('"%s"', $variable) : $variable;
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/Grid/Parser/OptionsParser.php]: (string) ???

File: src/Sylius/Bundle/ResourceBundle/Grid/Parser/OptionsParser.php [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    /**
+     * @param string $expression
+     * @param Request $request
+     *
+     * @return string
+     */
+    private function parseOptionExpression($expression, Request $request)
+    {
+        $expression = preg_replace_callback('/(\$\w+)/', function ($matches) use ($request) {
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/Grid/Parser/OptionsParser.php]:
```php
$expression = preg_replace_callback('/\$(\w+)/', function ($matches) use ($request) {
    $variable = $request->get($matches[1]);
    
    return is_string($variable) ? sprintf('"%s"', $variable) : $variable;
}, $expression);
```

This way we can get rid of `substr`.

File: src/Sylius/Bundle/ResourceBundle/Grid/Renderer/TwigGridRenderer.php [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+     * @param array $actionTemplates
+     */
+    public function __construct(
+        GridRendererInterface $gridRenderer,
+        \Twig_Environment $twig,
+        OptionsParserInterface $optionsParser,
+        array $actionTemplates = []
+    ) {
+        $this->gridRenderer = $gridRenderer;
+        $this->twig = $twig;
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/Grid/Renderer/TwigGridRenderer.php]: It's not even used.

Comment (GSadee) [inline src/Sylius/Bundle/ResourceBundle/Grid/Renderer/TwigGridRenderer.php]: Yes, it is here: https://github.com/Sylius/Sylius/pull/6793/files#diff-bb793d8d4c25918c7655a9d11a4a218eR96

**Comment by pjedrzejewski** [comment]
@GSadee Could you rebase, please? :) Thanks!

**Comment by pjedrzejewski** [comment]
Thanks Grzesiu!

**Comment by ARoddis** [comment]
Hmm this removed the possibility to "disable" Actions...is there another way now?

**Pull request merged**